tests/multi_device_test.py

```diff
@@ -269,6 +270,13 @@ def test_grad_device_put_src_inference(self):
     self.assert_committed_to_device(y, devices[1])
     self.assert_committed_to_device(x_bar, devices[0])
 
+  def test_lax_full_sharding(self):
```

Comment:
I don't think we need 2 tests? Just add a test in 1 place?

Reply:
These test two different cases: this one tests `sharding` with multiple devices, and the other tests sharding with a single device.

Reply:
I don't think we need to duplicate every test like this.

You can just add one test in a file where the cpu flag is set and tue single gpu config we run should take care of the single device case.

Reply:
OK, removed the multi_device version of this test.

Reply:
I meant keep the multi-device version of it so CPU will run with 8 devices and 1 GPU test in TAP will run with 1 device covering all the cases you want.